File: VB/AppointmentExporter.vb
Imports System
Imports System.Web.UI.WebControls
Imports Google.Apis.Services
Imports Google.Apis.Auth.OAuth2
Imports Google.Apis.Calendar.v3
Imports Google.Apis.Calendar.v3.Data
Imports DevExpress.XtraScheduler
Imports DevExpress.Web.ASPxScheduler
Imports System.Web.UI
Imports System.ComponentModel

Namespace GoogleCalendarAPI
    Public Class AppointmentExporter
        Private control As ASPxScheduler
        Private service As CalendarService
        Private calendarId As String
        Public Sub New(ByVal control As ASPxScheduler, ByVal service As CalendarService, ByVal calendarId As String)
            Me.control = control
            Me.service = service
            Me.calendarId = calendarId
        End Sub
        Private Sub AssignProperties(ByVal customEvent As Appointment, ByVal newEvent As [Event])
            newEvent.Summary = customEvent.Subject
            newEvent.Description = customEvent.Description
            newEvent.Location = customEvent.Location
            newEvent.Start = CalendarApiHelper.MakeEventDate(customEvent.Start)
            newEvent.Start.TimeZone = CalendarApiHelper.GetVTimeZone(control.Storage.TimeZoneId)
            newEvent.End = CalendarApiHelper.MakeEventDate(customEvent.End)
            newEvent.End.TimeZone = CalendarApiHelper.GetVTimeZone(control.Storage.TimeZoneId)
        End Sub

        Private Function CreateEvent(ByVal apt As Appointment) As [Event]
            Dim instance As New [Event]()
            AssignProperties(apt, instance)
            Return instance
        End Function
        Public Sub Export()
            For i As Integer = 0 To control.SelectedAppointments.Count - 1
                Dim apt As Appointment = control.SelectedAppointments(i)
                Dim instance As [Event] = CreateEvent(apt)
                If apt.Type = AppointmentType.Normal Then
                    Me.service.Events.Insert(instance, Me.calendarId).Execute()
                End If

            Next i
        End Sub
    End Class
End Namespace

File: CS/SchedulerExportGoogleCalendarWeb/WebSite/App_Code/AppointmentExporter.cs
using System;
using System.Web.UI.WebControls;
using Google.Apis.Services;
using Google.Apis.Auth.OAuth2;
using Google.Apis.Calendar.v3;
using Google.Apis.Calendar.v3.Data;
using DevExpress.XtraScheduler;
using DevExpress.Web.ASPxScheduler;
using System.Web.UI;
using System.ComponentModel;


public class AppointmentExporter
{
    ASPxScheduler control;
    CalendarService service;
    string calendarId;
    public AppointmentExporter(ASPxScheduler control, CalendarService service, string calendarId)
    {
        this.control = control;
        this.service = service;
        this.calendarId = calendarId;
    }
    private void AssignProperties(Appointment customEvent, Event newEvent)
    {
        newEvent.Summary = customEvent.Subject;
        newEvent.Description = customEvent.Description;
        newEvent.Location = customEvent.Location;
        newEvent.Start = CalendarApiHelper.MakeEventDate(customEvent.Start);
        newEvent.Start.TimeZone = CalendarApiHelper.GetVTimeZone(control.Storage.TimeZoneId);
        newEvent.End = CalendarApiHelper.MakeEventDate(customEvent.End);
        newEvent.End.TimeZone = CalendarApiHelper.GetVTimeZone(control.Storage.TimeZoneId);
    }

    private Event CreateEvent(Appointment apt)
    {
        Event instance = new Event();
        AssignProperties(apt, instance);
        return instance;
    }
    public void Export()
    {
        for (int i = 0; i < control.SelectedAppointments.Count; i++)
        {
            Appointment apt = control.SelectedAppointments[i];
            Event instance = CreateEvent(apt);
            if (apt.Type == AppointmentType.Normal)
            {
                this.service.Events.Insert(instance, this.calendarId).Execute();
            }

        }
    }
}
